### PR TITLE
Moved Controller, Entity and Form folders to the proper place

### PR DESCRIPTION
**Pull request by Sutharsan**
I moved files and folders inside lib/Drupal/foo_bar around according to the structure that core modules currently use:
  lib/Drupal/foo_bar
    Controller
    Entity
    Form

### CODE REVIEW

**Comment by Sutharsan** [comment]
Made some changes to the entity definition:
- Added 'translatable'. Since you have a translation controller definition, I assume you meant the entity to be translatable.
- Removed 'field_cache'. This is very rarely disabled and would be a bad example.
- Used an 'admin_permission' that was defined by the module and makes more sense.

File: lib/Drupal/foo_bar/Entity/FooBar.php [hunk 1]
```diff
@@ -16,19 +18,18 @@
  *   label = @Translation("FooBar entity"),
  *   controllers = {
  *     "view_builder" = "Drupal\foo_bar\Entity\FooBarViewBuilder",
- *     "list" = "Drupal\foo_bar\Entity\Controller\FooBarListController",
- *
+ *     "list" = "Drupal\foo_bar\Controller\FooBarListController",
```
Comment (Berdir) [inline lib/Drupal/foo_bar/Entity/FooBar.php]: Not sure about this. I think Controller is reserved for actual routing controllers, this is an entity controller that will soon be renamed to EntityListBuilder. So I kind of like to keep them unter the Entity namespace.

Same for entity forms, they're not standalone forms.

Core isn't doing that yet, it's more of a personal preference that I started to experiment with in my own contrib projects.

Moving the Settings form out of Entity makes a lot of sense, though

**Comment by Sutharsan** [comment]
Ah, Berdir you are behind this :) I did not really looked at the content of the Controller folder, now I see.

I was looking at a working entity example, so far it serves it purpose.

**Comment by christophegalli** [comment]
Once the experts agree, the humble documenter will adapt :-)

@Sutharsan: this module is the summary of this documentation piece: https://drupal.org/node/2192175

Thanks for the supprt, comments are welcome

**Comment by Sutharsan** [comment]
@christophegalli: That documentation page is how I ended up here. Thanks for making the code and updating the documentation.

**Comment by Berdir** [comment]
Kind of behind, @christophegalli is helping me to improve the documentation and I'm helping him with D8 so that he can help me ;)

**Comment by Sutharsan** [comment]
I reverted the move of the entity controllers, except for the settings form.

Further I changed the path to the "Foo Bar Settings" page from admin/configuration/foo_bar_settings to admin/structure/foo_bar_settings. Despite the path was "admin/configuration/..." the "Foo Bar Settings" link was still shown on "admin/structure". Either path should change or the link should not be there.

**Comment by christophegalli** [comment]
Thanks, merged your changes.
I will update the documentation accordingly.

**Pull request merged**